graphstorm-processing/docker/0.2.2/sagemaker/Dockerfile.cpu

```diff
@@ -11,32 +11,17 @@ ENV LC_ALL=C.UTF-8
 ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib"
 ENV LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/opt/conda/lib"
 ENV PATH=/opt/conda/bin:$PATH
+ENV PIP_NO_CACHE_DIR=1
 
-# Install GSProcessing requirements to pipenv Python
-RUN pipenv install \
-    boto3==1.28.38 \
-    joblib==1.3.1 \
-    mock==5.1.0 \
-    pandas==1.3.5 \
-    pip==23.1.2 \
-    protobuf==3.20.3 \
-    psutil==5.9.5 \
-    pyarrow==13.0.0 \
-    pyspark==3.4.1 \
-    scipy==1.11.3 \
-    setuptools \
-    transformers==4.37.1 \
-    spacy==3.6.0 \
-    torch==2.1.0 \
-    wheel \
-    && rm -rf /root/.cache
-# Do a pipenv sync so our base libs are independent from our editable code, making them cacheable
-RUN pipenv sync --system && python3 -m spacy download en_core_web_lg \
+WORKDIR /usr/lib/spark/code/
+
+# Install GSProcessing dependencies to system Python 3.9
+COPY requirements.txt requirements.txt
+RUN /usr/local/bin/python3.9 -m pip install --no-cache-dir -r /usr/lib/spark/code/requirements.txt \
```

Comment:
We may need transformers as we are going to support BERT.

Reply:
We have transformers here, as we already have the transformers installed in the requirements.txt. It is generated by poetry.

Reply:
Yes to clarify this `requirements.txt` file is [generated at build-time by poetry,](https://github.com/thvasilo/graphstorm/blob/c85b20404ae3bb564636f6bb3bcfc1a76290278b/graphstorm-processing/docker/build_gsprocessing_image.sh#L150-L151) based on the dependencies set in pyproject.toml (which include transformers).